src/Node.js

```diff
@@ -0,0 +1,121 @@
+  // ### DOM Node Shim
+  //
+  // This represents a superficial form of a DOM node which contains most of
+  // the properties that a regular DOM node would contain, but only the ones
+  // needed by slowparse.
+module.exports = (function(){
+  "use strict";
+
+  var voidHtmlElements = require("./voidHtmlElements");
+
+  function Node(nodeType, nodeName, nodeValue, namespaceURI) {
+    this.nodeType = nodeType;
+    this._nodeName = nodeName;
+    this.nodeValue = nodeValue || "";
+    this.namespaceURI = namespaceURI || "";
+    this.parentNode = false;
+    this.nextSibling = false;
+    this.isVoid = voidHtmlElements.indexOf(nodeName) > -1;
+    this.childNodes = [];
+    this.attributes = [];
+    this._attributeMap = {};
+  }
+
+  Node.prototype = {
+    // Add a node to the current node as a child
+    appendChild: function(node) {
+      var lastChild = this.childNodes.slice(-1)[0];
+      if(lastChild) {
+        lastChild.nextSibling = node;
+      }
+
+      node.parentNode = this;
+      this.childNodes.push(node);
+    },
+    // Create a deep copy of the current node
+    cloneNode: function() {
+      var node = new Node(this.nodeType, this.nodeName, this.nodeValue, this.namespaceURI);
+      node.nextSibling = this.nextSibling ? this.nextSibling.cloneNode() : false;
+      node.isVoid = this.isVoid;
+      node.childNodes = this.childNodes.map(function(childNode) {
+        return childNode.cloneNode();
+      });
+      node.attributes = this.attributes.map(function(attribute) {
+        return attribute.cloneNode();
```

Comment:
given the overloading here to have `cloneNode` act on nodes *and* attributes, I'd recommend simply calling the function `clone()` instead of `cloneNode`, so we don't mislead future code-readers/contributors/devs